rust/scx_rustland_core/assets/bpf/main.bpf.c

```diff
@@ -433,15 +433,23 @@ static s32 pick_idle_cpu(struct task_struct *p, s32 prev_cpu)
 	 * CPU (ignore if this cpumask completely overlaps with the task's
 	 * cpumask).
 	 */
-	bpf_cpumask_and(l2_mask, p->cpus_ptr, cast_mask(l2_domain));
+	if (bpf_cpumask_and(l2_mask, p->cpus_ptr, cast_mask(l2_domain)) < 0){
+		l2_mask = NULL;
+		cpu = -ENOENT;
+		goto out_put_cpumask;
```

Comment:
Even if l2_mask is empty we still want to try with the l3_mask and all the other stuff below, we don't want to give up immediately.